### PR TITLE
acl: Make some code refactoring

### PR DESCRIPTION
**Pull request by cthulhu-rider**
Also fix `AccessControlPolicy` XML.

### CODE REVIEW

File: api/handler/acl.go [hunk 1]
```diff
@@ -56,13 +56,17 @@ const (
 	s3GetObjectVersion           = "s3:GetObjectVersion"
 )
 
-// AWSACL is aws permission constants.
-type AWSACL string
```
Comment (roman-khimov) [inline api/handler/acl.go:60]: Doesn't seem to be very useful.

File: api/handler/acl.go [hunk 2]
```diff
@@ -56,13 +56,17 @@ const (
...
+type amazonS3Permission string
 
+// various access permissions defined in
+// https://docs.aws.amazon.com/AmazonS3/latest/API/API_Grant.html.
+//
+// See also https://docs.aws.amazon.com/AmazonS3/latest/userguide/acl-overview.html#permissions.
 const (
-	aclFullControl AWSACL = "FULL_CONTROL"
-	aclWrite       AWSACL = "WRITE"
-	aclRead        AWSACL = "READ"
```
Comment (roman-khimov) [inline api/handler/acl.go:65]: Especially these renames.

File: api/handler/acl.go [hunk 3]
```diff
@@ -69,13 +69,15 @@ const (
 	awsPermRead        amazonS3Permission = "READ"
 )
 
-// GranteeType is aws grantee permission type constants.
-type GranteeType string
```
Comment (roman-khimov) [inline api/handler/acl.go:69]: Can also be omitted, s3-gw is not something that is supposed to be reused package-by-package.

File: api/handler/acl.go (hunk 2 above)
Comment (cthulhu-rider) [reply api/handler/acl.go:65]: I can't understand what `acl` stands for - Amazon S3 or NeoFS.

File: api/handler/acl.go (hunk 1 above)
Comment (cthulhu-rider) [reply api/handler/acl.go:60]: It's not AWS ACL, it's permissions. Also it is unused outside the package, and has couple of unexported constants. I wanna realize all these things from the definition.

File: api/handler/acl.go (hunk 3 above)
Comment (cthulhu-rider) [reply api/handler/acl.go:69]: Then there is no sense to export these things.

File: api/handler/acl.go (hunk 1 above)
Comment (cthulhu-rider) [reply api/handler/acl.go:60]: Btw all values are defined in https://pkg.go.dev/github.com/aws/aws-sdk-go/service/s3#pkg-constants. I don't know why we define them locally.

**Review by KirillovDenis — APPROVED** [review]

File: api/handler/acl.go (hunk 1 above)
Comment (roman-khimov) [reply api/handler/acl.go:60]: I guess the main problem is that this package has _a lot_ more inside.

**Review by roman-khimov — APPROVED** [review]
In general, please avoid renamings unless they're strictly required. Still, this improves things, so can be accepted.

**Pull request merged**